### PR TITLE
add log cert controller

### PR DESCRIPTION
**Pull request by zhiweiyin318**

### CODE REVIEW

File: pkg/controllers/clusterinfo/clusterinfo_controller.go [hunk 1]
```diff
@@ -171,7 +221,18 @@ func (r *ClusterInfReconciler) newClusterInfoByManagedCluster(cluster *clusterv1
 		},
 		Spec: clusterinfov1beta1.ClusterInfoSpec{
 			MasterEndpoint: endpoint,
-			LoggingCA:      r.caData,
+			LoggingCA:      r.getLogCA(),
 		},
 	}
 }
+
+func (r *ClusterInfReconciler) getLogCA() []byte {
```
Comment (qiujian16) [inline pkg/controllers/clusterinfo/clusterinfo_controller.go:229]: I think you should return error here.

File: pkg/controllers/certrotation/certrotation_controller.go [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+		certSecret.Data = map[string][]byte{}
+		if err := newCertSecret(certSecret, signerName, defaultValidity); err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if err := r.client.Create(ctx, certSecret); err != nil {
+			return ctrl.Result{}, err
+		}
+
+	case err == nil:
```
[the remark this answers is not in the document]
Comment (zhiweiyin318) [reply pkg/controllers/certrotation/certrotation_controller.go]: done

File: pkg/controllers/clusterinfo/clusterinfo_controller.go (hunk 1 above)
Comment (zhiweiyin318) [reply pkg/controllers/clusterinfo/clusterinfo_controller.go]: done

File: pkg/controllers/clusterinfo/clusterinfo_controller.go [hunk 3]
```diff
@@ -30,31 +33,47 @@ const (
...
-func SetupWithManager(mgr manager.Manager, caData []byte) error {
-	if err := add(mgr, newClusterInfoReconciler(mgr, caData)); err != nil {
+func SetupWithManager(mgr manager.Manager, logCertSecret string) error {
+	var err error
+	logCertSecretNamespace, logCertSecretName, err = cache.SplitMetaNamespaceKey(logCertSecret)
+	if err != nil {
+		return err
+	}
+	if logCertSecretNamespace == "" {
+		nsBytes, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
```
[the remark this answers is not in the document]
Comment (zhiweiyin318) [reply pkg/controllers/clusterinfo/clusterinfo_controller.go]: done

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**